src/Campaigns/Blocks/CampaignGoal/app.tsx

```diff
@@ -0,0 +1,26 @@
+import {render} from '@wordpress/element';
+import useCampaign from '../shared/hooks/useCampaign';
+import App from './app/index';
+
+const BlockApp = ({campaignId}: { campaignId: number }) => {
+    const {campaign, hasResolved} = useCampaign(campaignId);
+
+    if (!hasResolved || !campaignId) {
+        return null;
+    }
+
+    return <App campaign={campaign} />;
+}
+
+/**
+ * @unreleased
+ */
+const nodeList = document.querySelectorAll('.give-campaigns-goalBlock-container');
```

Comment:
We should go with the data attribute approach here too 😄 

`data-givewp-goal`

Reply:
Resolved https://github.com/impress-org/givewp/pull/7702/commits/b85d99badf275fb0d12385d03b53beb46f696f66